Source/Core/DiscIO/DirectoryBlob.h

```diff
@@ -196,8 +196,7 @@ class DirectoryBlobPartition
       const std::function<void(std::vector<FSTBuilderNode>* fst_nodes, FSTBuilderNode* dol_node)>&
           fst_callback);
 
-  // We do not allow copying, because it might mess up the pointers inside DiscContents
```

Comment:
You haven't solved the problem of the pointers inside DiscContents. (More specifically, the problem is when `ContentSource` is a pointer.)

Reply:
Perhaps consider using `std::shared_ptr<std::vector<u8>>` instead of raw pointers to the contents of an `std::vector<u8>`.